publication-rest/src/main/java/no/unit/nva/publication/fetch/FetchPublicationHandler.java

```diff
@@ -99,21 +105,22 @@ private static void temporaryHackWhileWeWaitForUnauthenticatedUserAccess(Publica
         }
     }
 
-    @Override
-    protected Integer getSuccessStatusCode(Void input, String output) {
-        return statusCode;
-    }
-
     @JacocoGenerated
     private static ResourceService defaultResourceService(AmazonDynamoDB client) {
         return new ResourceService(client, CLOCK);
     }
 
+    private boolean publicationIsLogicallyDeleted(Publication publication) {
+        return PublicationStatus.DELETED.equals(publication.getStatus());
+    }
+
     private String createResponse(RequestInfo requestInfo,
-                                  Publication publication)
-        throws UnsupportedAcceptHeaderException {
+                                  Publication publication) throws UnsupportedAcceptHeaderException, GoneException {
         String response = null;
         var contentType = getDefaultResponseContentTypeHeaderValue(requestInfo);
+        if (publicationIsLogicallyDeleted(publication)) {
+            createGoneException();
```

Comment:
Personally, I would say createGoneException() method is not needed. It is not reused, and is a one-liner anyway.

Reply:
Fikset